examples/src/main/java/ai/djl/examples/inference/TextGeneration.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ * with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package ai.djl.examples.inference;
+
+import ai.djl.MalformedModelException;
+import ai.djl.engine.Engine;
+import ai.djl.ndarray.NDArray;
+import ai.djl.ndarray.NDList;
+import ai.djl.ndarray.NDManager;
+import ai.djl.ndarray.types.DataType;
+import ai.djl.ndarray.types.Shape;
+import ai.djl.repository.zoo.ModelNotFoundException;
+import ai.djl.translate.CausalLMOutput;
+import ai.djl.translate.LMAdapter;
+
+import java.io.IOException;
+
+public final class TextGeneration {
+
+    private TextGeneration() {}
+
+    public static void main(String[] args) {
+        mainOnnx(args);
+        mainPt(args);
+    }
+
+    public static void mainOnnx(String[] args) {
```

Comment:
I'm noticing that these examples are almost identical. Is it possible to refactor out a common helper. So, each one would just initialize the `LMAdapter` and then pass it into the helper. Maybe the helper would look something like `public static void generateInternal(LMAdapter generator);`